view/frontend/web/insights.js

```diff
@@ -0,0 +1,250 @@
+requirejs([
+    'jquery',
+    'algoliaAnalytics',
+    'algoliaBundle',
+], function ($, algoliaAnalytics, algoliaBundle) {
+
+    var algoliaInsights = {
+        config: null,
+        defaultIndexName: null,
+
+        track: function(algoliaConfig) {
+
+            this.config = algoliaConfig;
+            this.defaultIndexName = algoliaConfig.indexName + '_products';
+
+            if (algoliaConfig.ccAnalytics.enabled
+                || algoliaConfig.personalization.enabled) {
+
+                this.initializeAnalytics();
+                this.addSearchParameters();
+                this.bindData();
+                this.bindEvents();
+            }
+        },
+
+        initializeAnalytics: function() {
+            algoliaAnalytics.init({
+                appId: this.config.applicationId,
+                apiKey: this.config.apiKey
+            });
+
+            var userAgent = 'insights-js-in-magento (' + this.config.extensionVersion + ')';
```

Comment:
what's `this.config.extensionVersion` ? is it refering to the version of the `algoliasearch-magento-2` or something else?

Reply:
Yes, this is to add the extension version to the userAgent!